components/places/ios/Places/Bookmark.swift

```diff
@@ -24,3 +24,246 @@ public enum BookmarkRoots {
         BookmarkRoots.UnfiledFolderGUID,
     ])
 }
+
+// Keeping `BookmarkNodeType` in the swift wrapper because the iOS code relies on the raw value of the variants of
+// this enum.
+public enum BookmarkNodeType: Int32 {
+    // Note: these values need to match the Rust BookmarkType
+    // enum in types.rs
+    case bookmark = 1
+    case folder = 2
+    case separator = 3
+    // The other node types are either queries (which we handle as
+    // normal bookmarks), or have been removed from desktop, and
+    // are not supported
+}
+
+/**
+ * A base class containing the set of fields common to all nodes
+ * in the bookmark tree.
+ */
+public class BookmarkNodeData {
```

Comment:
I added these classes back to decrease the size of the changeset in the iOS PR, but I definitely think this is something that could be refactored in the future--just not convinced we need to take that on now. Feel free to challenge me on this.